Wobble/Scheduling/TaskHandler.cs

```diff
@@ -32,6 +32,16 @@ public class TaskHandler<T, TResult> : IDisposable
         /// </summary>
         public event EventHandler<TaskCancelledEventArgs<T>> OnCancelled;
 
+        /// <summary>
+        ///     Event Invoked when the task gets run.
+        /// </summary>
+        public event EventHandler<TaskStartedEventArgs<T>> OnStarted;
+
+        /// <summary>
+        ///    SemaphoreSlim used to sync tasks with a delay.
+        /// </summary>
+        private SemaphoreSlim SemaphoreSlim { get; } = new SemaphoreSlim(1, 1);
```

Comment:
From what I understand, a `Semaphore` with 1 initial and 1 maximum count, as used in this class, is exactly equivalent to a mutex (so a `lock {}` around the `try {} catch {}` statement inside `TaskRun()`, with `TaskRun()` ran in a separate thread manually rather than via `async`).